medusa/providers/nzb/binsearch.py

```diff
@@ -41,6 +52,230 @@ def __init__(self):
+                    continue
+                except Exception as error:
+                    continue
+
+                if not all([title, nzb_id]):
+                    continue
+                # Obtain the size from the 'description'
+                size_field = BinSearchProvider.size_regex.search(col['subject'].text)
+                if size_field:
+                    size_field = size_field.group(1)
+                size = convert_size(size_field, sep='\xa0') or -1
+
+                query = query or title
+
+                download_url = 'https://www.binsearch.info/fcgi/nzb.fcgi?q={query}&max=100&adv_age=1100&server='.format(
+                    query=query
+                )
+
+                download_url = '{download_url}|nzb_id={nzb_id}'.format(download_url=download_url, nzb_id=nzb_id)
+
+                # For future use
+                # detail_url = 'https://www.binsearch.info/?q={0}'.format(title)
+
+                date = col['age'].get_text(strip=True)
+
+                pubdate_raw = date
+                pubdate = self.parse_pubdate(pubdate_raw, human_time=True)
+
+                item = {
+                    'title': title,
+                    'link': download_url,
+                    'size': size,
+                    'pubdate': pubdate,
+                }
+                if mode != 'RSS':
+                    log.debug('Found result: {0}', title)
+
+                items.append(item)
+
+        return items
+
+    def download_result(self, result):
+        """Download result from provider."""
+        if not self.login():
+            return False
+
+        urls, filename = self._make_url(result)
+
+        for url in urls:
+            if 'NO_DOWNLOAD_NAME' in url:
+                continue
+
+            if url.startswith('http'):
+                self.headers.update({
+                    'Referer': '/'.join(url.split('/')[:3]) + '/'
+                })
+
+            log.info('Downloading {result} from {provider} at {url}',
+                     {'result': result.name, 'provider': self.name, 'url': url})
+
+            verify = False if self.public else None
+
+            url, data = url.split('|')
+
+            data = {
+                data.split('=')[1]: 'on',
+                'action': 'nzb'
+            }
+
+            if self.download_file(url, data, filename, session=self.session, headers=self.headers,
+                                  hooks={'response': self.get_url_hook}, verify=verify):
+
+                if self._verify_download(filename):
+                    log.info('Saved {result} to {location}',
+                             {'result': result.name, 'location': filename})
+                    return True
+
+        if urls:
+            log.warning('Failed to download any results for {result}',
+                        {'result': result.name})
+
+        return False
+
+    def download_file(self, url, data, filename, session=None, headers=None, **kwargs):
```

Comment:
Then only real change here, is adding the data paramater, and changing the request to a POST. The entire method can be removed if we add some params to the original download_files.